src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/AndInstruction.cs

```diff
@@ -159,13 +159,13 @@ public override int Run(InterpretedFrame frame)
                     }
                     else
                     {
-                        frame.Push((bool)right ? null : ScriptingRuntimeHelpers.Boolean_False);
+                        frame.Push((bool)right ? null : Utils.BoxedFalse);
```

Comment:
How about we add a Debug.Assert to the frame.Push overload that takes an object?  It could check to see if the input ```is bool```, and if it is, verify reference equality with the BoxedTrue or BoxedFalse inputs.  That'll help us spot current or future misuse.